ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -509,6 +533,7 @@ encodeShelleyResult query = case query of
     GetStakeDistribution         -> toCBOR
     GetFilteredUTxO {}           -> toCBOR
     GetUTxO                      -> toCBOR
+    GetCurrentLedgerState        -> wrapCBORinCBOR encode
```

Comment:
Can't you just use `toCBOR` instead of `encode` (and `fromCBOR` instead of `decode`)? Then you won't need all these orphans below.

Reply:
Let also add a comment saying why we use CBOR-in-CBOR for this one.

Reply:
Fixed.

Reply:
Why not `wrapCBORinCBOR toCBOR`? (Or does that not work?)

Reply:
It does not work.